lib/letsencrypt.js

```diff
@@ -36,180 +39,81 @@ function createServer(app, settings, certificates, leSettings, log) {
+                            severityFn = log.error;
+                            break;
+                        default:
+                            severityFn = log.debug;
+                    }
+                    severityFn(details);
                 }
+            };
+            if (settings.leStaging) {
+                log.debug('Greenlock staging enabled');
+                glOpts.staging = true;
             }
-
-            const lex = LE.create({
-                version:         'v02',
-                debug:           true,
-                configDir:       leDir,
-                agreeTos:        true,
-                store:           require('./letsencryptStore.js').create({debug: true, log: leLog}),
-                server:          'https://acme-v02.api.letsencrypt.org/directory', //'staging',
-                email:           leSettings.email,
-                approvedDomains: leSettings.domains,
-                log:             leLog
-            });
-
-            if (settings.leUpdate) {
-                settings.lePort = parseInt(settings.lePort, 10) || 80;
-                // handles acme-challenge and redirects to https
-                // used for validation of requests like  http://example.com/.well-known/acme-challenge/BLABALBAL
-                try {
-                    const server = require('http').createServer(lex.middleware(function redirectHttps(req, res) {
-                        res.setHeader('Location', `https://${req.headers.host}:${settings.port}${req.url}`);
-                        res.statusCode = 302;
-                        res.end('<!-- Hello Developer Person! Please use HTTPS instead -->');
-                    }));
-
-                    server.on('error', e => {
-                        if (e.toString().includes('EACCES') && settings.lePort <= 1024) {
-                            log.error(`node.js process has no rights to start LetsEncrypt challenge server on port ${settings.lePort}.\n` +
-                                `Do you know that on linux you need special permissions for ports under 1024?\n` +
-                                `You can call in shell following script to allow it for node.js: "iobroker fix"`
-                            );
-                        } else {
-                            log.error(`LetsEncrypt challenge server could not be started on port ${settings.lePort}: ${e}`);
-                        }
-                    });
-
-                    server.listen(settings.lePort, () => {
-                        log.info('LetsEncrypt challenge server is started on ' + settings.lePort);
-                    });
-                } catch (err) {
-                    log.error('LetsEncrypt challenge server could not be started: ' + err);
+            const gl = require('greenlock').create(glOpts);
+            gl.manager.defaults({
+                agreeToTerms: true,
+                subscriberEmail: leSettings.email
+            }).then(() => {
```

Comment:
I would prefer `await` wrapped in try-catch for newly added code, instead of `then` `catch`

Reply:
I don't like async too. It is just sugar, but its hide the real mechanism.

Reply:
> but its hide the real mechanism

Yeah, that's the point. The "real mechanism" has been shoehorned into a language that wasn't made for it. With `await` you can at least read the code and not worry about it.

Reply:
The `.catch` is below. I don't see a compelling reason to wrap this entire thing in a separate `try... catch`.

Reply:
@foxriver76 meant converting the Promise.then().catch() into `try { await ... } catch (e) {...}`

Reply:
OK, yes. This way we can make the whole function `async` instead of explicitly returning a promise. This will make things cleaner.

Reply:
That is cleaner: https://github.com/ioBroker/ioBroker.js-controller/pull/1159/commits/121192a0f4ac4499f91f78897da17dd45e031a86